### PR TITLE
Bug 2126694: add filter for kms connection details

### PR DESCRIPTION
**Pull request by SanjalKatiyar**

### CODE REVIEW

File: packages/ocs/storage-class/sc-form.tsx [hunk 1]
```diff
@@ -392,16 +395,20 @@ const ExistingKMSDropDown: React.FC<ExistingKMSDropDownProps> = ({
           try {
             // removing any object having syntax error
             // or, which are not supported by UI.
-            const kmsData = JSON.parse(connectionDetails);
+            const kmsData: Object = JSON.parse(connectionDetails);
```
Comment (GowthamShanmugam) [inline packages/ocs/storage-class/sc-form.tsx:398]:
```suggestion
            const kmsData: KMSConfigMap = JSON.parse(connectionDetails);
```

File: packages/odf/constants/kms.ts [hunk 2]
```diff
@@ -32,6 +33,15 @@ export const SupportedProviders = {
   [ProviderNames.THALES]: {
     group: 'Thales (using KMIP)',
     supported: [KmsImplementations.KMIP],
+    filter: (kms: Object, secretsList: SecretKind[]) => {
```
Comment (GowthamShanmugam) [inline packages/odf/constants/kms.ts:36]:
```suggestion
    filter: (kms: ThalesConfigMap, secretsList: SecretKind[]) => {
```

File: packages/ocs/storage-class/sc-form.tsx (hunk 1 above)
Comment (SanjalKatiyar) [reply packages/ocs/storage-class/sc-form.tsx:398]: it is not exactly the ConfigMap, but "data" part of the configmap which is a JSON string and we are converting to JS Object...

File: packages/odf/constants/kms.ts (hunk 2 above)
Comment (GowthamShanmugam) [inline packages/odf/constants/kms.ts:36]: Shall we add the proper type for this object?
something like
export const SupportedProviders : SupportedProvidersProps

Comment (SanjalKatiyar) [reply packages/odf/constants/kms.ts:36]: ack... will add that...

File: packages/ocs/storage-class/sc-form.tsx (hunk 1 above)
Comment (SanjalKatiyar) [reply packages/ocs/storage-class/sc-form.tsx:398]: I can check and create a new type for it... let me check...

File: packages/ocs/storage-class/sc-form.tsx [hunk 3]
```diff
@@ -506,19 +527,15 @@ export const StorageClassEncryptionKMSID: React.FC<ProvisionerProps> = ({
...
-    namespace: CEPH_STORAGE_NAMESPACE,
-    name: KMSConfigMapCSIName,
-  };
   const [infra, infraLoaded, infraLoadError] = useK8sGet<any>(
     InfrastructureModel,
     'cluster'
   );
   const [csiConfigMap, csiConfigMapLoaded, csiConfigMapLoadError] =
     useK8sWatchResource<ConfigMapKind>(csiCMWatchResource);
+  const [secrets, secretsLoaded, secretsLoadError] =
```
Comment (GowthamShanmugam) [inline packages/ocs/storage-class/sc-form.tsx:536]: shall we add useK8sWatchResources for this, do you think any performance challenges it there if we use useK8sWatchResources?

Comment (SanjalKatiyar) [reply packages/ocs/storage-class/sc-form.tsx:540]: Is there any advantage of using one over another ?? I think both will work similarly...

Comment (GowthamShanmugam) [reply packages/ocs/storage-class/sc-form.tsx:540]: no advantage, just for better code maintenance

**Review by GowthamShanmugam — APPROVED** [review]

**Pull request merged**